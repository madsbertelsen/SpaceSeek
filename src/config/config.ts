import { feature } from 'topojson-client';
const countries = require('../assets/countries-50m.json');
export const COUNTRIES = feature(countries, countries.objects.countries)
  .features;

interface AppConfig {
  env: 'prod' | 'test';
  url: string;
  helpURL: string;
  spaceAPIURL: string;
}

export const config: AppConfig = {
  env: 'test',
  url: 'https://mariusreimer.com',
  helpURL: 'https://github.com/reime005/SpaceSeek/issues',
  spaceAPIURL: 'https://thespacedevs.com',
};
