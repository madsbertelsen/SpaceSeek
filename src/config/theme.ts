import { DefaultTheme } from 'styled-components/native';
import { BadgeType } from '../components/Badges/BadgeWrapper';

export type FontColor = 'regular' | 'subtle' | 'light' | 'primary';

declare module 'styled-components' {
  export interface DefaultTheme {
    headerBackgroundColor: string;
    backgroundColor: string;
    backgroundColorLight: string;
    successColor: string;
    failColor: string;
    listItemBackgroundColor: string;
    placeHolderFontColor: string;
    splashBackgroundColor: string;
    primaryColor: string;
    secondaryColor: string;
    searchBackgroundColor: string;
    focusedIconColor: string;
    unFocusedIconColor: string;
    mainFont: string;
    fontColors: { [index in FontColor]: string };
    badgeBackgroundColors: { [index in BadgeType]: string };
    badgeFontColors: { [index in BadgeType]: string };
  }
}

export const lightTheme: DefaultTheme = {
  headerBackgroundColor: '#fff',
  backgroundColor: '#fff',
  backgroundColorLight: 'rgb(250, 250, 250)',
  successColor: '#008214',
  failColor: '#990000',
  splashBackgroundColor: 'white',
  primaryColor: '#e96232',
  focusedIconColor: '#e96232',
  unFocusedIconColor: '#cccccc',
  placeHolderFontColor: '#cccccc',
  secondaryColor: '#666',
  listItemBackgroundColor: 'rgb(252,252,252)',
  searchBackgroundColor: 'rgb(252,252,252)',
  mainFont: '#090909',
  fontColors: {
    light: '#fff',
    primary: '#e96232',
    regular: '#090909',
    subtle: '#666',
  },
  badgeBackgroundColors: {
    wiki: '#e4e4e4',
    youtube: '#c93a2f',
    launch: '#dcf0f0',
  },
  badgeFontColors: {
    wiki: '#4c4c4c',
    youtube: '#fefcff',
    launch: '#090909',
  },
};

export const darkTheme: DefaultTheme = {
  headerBackgroundColor: '#0d0d0d',
  backgroundColor: '#0d0d0d',
  backgroundColorLight: '#8f8f8f',
  successColor: '#008214',
  failColor: '#990000',
  splashBackgroundColor: '#21202e',
  primaryColor: '#e96232',
  focusedIconColor: '#e96232',
  unFocusedIconColor: '#cccccc',
  placeHolderFontColor: '#fff',
  secondaryColor: '#cacaca',
  listItemBackgroundColor: '#1f1f1f',
  searchBackgroundColor: '#999999',
  mainFont: '#fff',
  fontColors: {
    light: '#fff',
    primary: '#e96232',
    regular: '#fff',
    subtle: '#fafafa',
  },
  badgeBackgroundColors: {
    wiki: '#e4e4e4',
    youtube: '#c93a2f',
    launch: '#dcf0f0',
  },
  badgeFontColors: {
    wiki: '#4c4c4c',
    youtube: '#fefcff',
    launch: '#090909',
  },
};
