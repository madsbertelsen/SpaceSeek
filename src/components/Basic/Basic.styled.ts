import styled from 'styled-components/native';
import * as ReactNative from 'react-native';
import { FontColor } from '../../config/theme';

export type Font = 'RobotoCondensed-Regular';

type FontSize = 'xs' | 's' | 'm' | 'l' | 'xl' | 'xxl' | 'super' | 'max';

export const fontSizes: { [index in FontSize]: string } = {
  xs: '10px',
  s: '12px',
  m: '14px',
  l: '16px',
  xl: '18px',
  xxl: '24px',
  super: '32px',
  max: '54px',
};

export const lineHeights: { [index in FontSize]: string } = {
  xs: '12px',
  s: '13px',
  m: '21px',
  l: '24px',
  xl: '27px',
  xxl: '36px',
  super: '48px',
  max: '81px',
};

type FontType = 'regular' | 'bold' | 'italic';

const fontFamilies: { [index in FontType]: string } = {
  regular: 'RobotoCondensed-Regular',
  bold: 'RobotoCondensed-Bold',
  italic: 'RobotoCondensed-Italic',
};

export interface IStyledCustomText {
  size: FontSize;
  fontType: FontType;
  fontColor: FontColor;
  backgroundColor: string;
}

export const StyledCustomText = styled.Text<IStyledCustomText>`
  color: ${(props) => props.theme.fontColors[props.fontColor]};
  backgroundcolor: ${(props) => props.backgroundColor};
  font-size: ${(props) => fontSizes[props.size]};
  line-height: ${(props) => lineHeights[props.size]};
  font-family: ${(props) => fontFamilies[props.fontType]};
`;

export const StyledBaseScroll = styled.ScrollView`
  flex: 1;
  width: 100%;
  height: 100%;
  background-color: ${({ theme }) => theme.backgroundColor};
`;

export const BasePage = styled(ReactNative.Animated.View)`
  flex: 1;
  width: 100%;
  background-color: ${({ theme }) => theme.backgroundColor};
`;

export const IconWrapper = styled.View`
  padding: 10px;
  border-radius: 8px;
  background-color: ${({ theme }) => theme.primaryColor};
`;
