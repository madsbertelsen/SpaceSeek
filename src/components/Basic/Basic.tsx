import React from 'react';
import * as ReactNative from 'react-native';

import * as S from './Basic.styled';

interface IRegularText
  extends ReactNative.TextProps,
    Partial<S.IStyledCustomText> {
  children: string | string[];
}

export const RegularText = (props: IRegularText) => {
  return (
    <S.StyledCustomText
      {...{ fontColor: 'regular', fontType: 'regular', size: 'm', ...props }}
    />
  );
};

export const Title = (props: IRegularText) => {
  return (
    <S.StyledCustomText
      {...{ fontColor: 'regular', fontType: 'bold', size: 'l', ...props }}
    />
  );
};

export const SubText = (props: IRegularText) => {
  return (
    <S.StyledCustomText
      {...{ fontColor: 'subtle', fontType: 'regular', size: 's', ...props }}
    />
  );
};

interface BaseScrollProps extends ReactNative.ScrollViewProps {
  children: React.ReactNode | React.ReactNode[];
}

export const BaseScroll = (props: BaseScrollProps) => {
  return (
    <S.StyledBaseScroll
      showsVerticalScrollIndicator={false}
      alwaysBounceVertical={false}
      contentContainerStyle={{ padding: 16 }}
      {...props}>
      {props.children}
    </S.StyledBaseScroll>
  );
};

export const BasePage = (props: any) => {
  return <S.BasePage {...props}>{props.children}</S.BasePage>;
};
