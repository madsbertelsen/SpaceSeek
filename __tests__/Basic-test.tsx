/**
 * @format
 */

import 'react-native';
import React from 'react';

import 'jest-styled-components/native';

import renderer from 'react-test-renderer';

import { RegularText, Title, SubText } from '../src/components/Basic/Basic';

import {
  BasePage,
  StyledBaseScroll,
  IconWrapper,
} from '../src/components/Basic/Basic.styled';

import { darkTheme, lightTheme } from '../src/config/theme';

it('Regular Text renders correctly', async () => {
  const tree = renderer
    .create(<RegularText theme={darkTheme}>TEST</RegularText>)
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('color', darkTheme.fontColors.regular);
  expect(tree).toHaveStyleRule('fontSize', 14);
});

it('Title renders correctly', async () => {
  const tree = renderer.create(<Title theme={lightTheme}>TEST</Title>).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('color', lightTheme.fontColors.regular);
  expect(tree).toHaveStyleRule('fontSize', 16);
});

it('SubText renders correctly', async () => {
  const tree = renderer
    .create(<SubText theme={lightTheme}>TEST</SubText>)
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('color', lightTheme.fontColors.subtle);
  expect(tree).toHaveStyleRule('fontSize', 12);
});

it('BasePage renders correctly', async () => {
  const tree = renderer.create(<BasePage theme={lightTheme} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('StyledBaseScroll renders correctly', async () => {
  const tree = renderer
    .create(<StyledBaseScroll theme={lightTheme} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('IconWrapper renders correctly', async () => {
  const tree = renderer.create(<IconWrapper theme={lightTheme} />).toJSON();
  expect(tree).toMatchSnapshot();
});
