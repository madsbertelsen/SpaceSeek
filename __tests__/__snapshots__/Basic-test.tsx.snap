// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasePage renders correctly 1`] = `
<View
  collapsable={false}
  nativeID="animatedComponent"
  style={
    Object {
      "backgroundColor": "#fff",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "width": "100%",
    }
  }
  theme={
    Object {
      "backgroundColor": "#fff",
      "backgroundColorLight": "rgb(250, 250, 250)",
      "badgeBackgroundColors": Object {
        "launch": "#dcf0f0",
        "wiki": "#e4e4e4",
        "youtube": "#c93a2f",
      },
      "badgeFontColors": Object {
        "launch": "#090909",
        "wiki": "#4c4c4c",
        "youtube": "#fefcff",
      },
      "failColor": "#990000",
      "focusedIconColor": "#e96232",
      "fontColors": Object {
        "light": "#fff",
        "primary": "#e96232",
        "regular": "#090909",
        "subtle": "#666",
      },
      "headerBackgroundColor": "#fff",
      "listItemBackgroundColor": "rgb(252,252,252)",
      "mainFont": "#090909",
      "placeHolderFontColor": "#cccccc",
      "primaryColor": "#e96232",
      "searchBackgroundColor": "rgb(252,252,252)",
      "secondaryColor": "#666",
      "splashBackgroundColor": "#21202e",
      "successColor": "#008214",
      "unFocusedIconColor": "#cccccc",
    }
  }
/>
`;

exports[`IconWrapper renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#e96232",
        "borderRadius": 8,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      },
    ]
  }
  theme={
    Object {
      "backgroundColor": "#fff",
      "backgroundColorLight": "rgb(250, 250, 250)",
      "badgeBackgroundColors": Object {
        "launch": "#dcf0f0",
        "wiki": "#e4e4e4",
        "youtube": "#c93a2f",
      },
      "badgeFontColors": Object {
        "launch": "#090909",
        "wiki": "#4c4c4c",
        "youtube": "#fefcff",
      },
      "failColor": "#990000",
      "focusedIconColor": "#e96232",
      "fontColors": Object {
        "light": "#fff",
        "primary": "#e96232",
        "regular": "#090909",
        "subtle": "#666",
      },
      "headerBackgroundColor": "#fff",
      "listItemBackgroundColor": "rgb(252,252,252)",
      "mainFont": "#090909",
      "placeHolderFontColor": "#cccccc",
      "primaryColor": "#e96232",
      "searchBackgroundColor": "rgb(252,252,252)",
      "secondaryColor": "#666",
      "splashBackgroundColor": "#21202e",
      "successColor": "#008214",
      "unFocusedIconColor": "#cccccc",
    }
  }
/>
`;

exports[`Regular Text renders correctly 1`] = `
<Text
  fontColor="regular"
  fontType="regular"
  size="m"
  style={
    Array [
      Object {
        "color": "#fff",
        "fontFamily": "RobotoCondensed-Regular",
        "fontSize": 14,
        "lineHeight": 21,
      },
    ]
  }
  theme={
    Object {
      "backgroundColor": "#0d0d0d",
      "backgroundColorLight": "#8f8f8f",
      "badgeBackgroundColors": Object {
        "launch": "#dcf0f0",
        "wiki": "#e4e4e4",
        "youtube": "#c93a2f",
      },
      "badgeFontColors": Object {
        "launch": "#090909",
        "wiki": "#4c4c4c",
        "youtube": "#fefcff",
      },
      "failColor": "#990000",
      "focusedIconColor": "#e96232",
      "fontColors": Object {
        "light": "#fff",
        "primary": "#e96232",
        "regular": "#fff",
        "subtle": "#fafafa",
      },
      "headerBackgroundColor": "#0d0d0d",
      "listItemBackgroundColor": "#1f1f1f",
      "mainFont": "#fff",
      "placeHolderFontColor": "#fff",
      "primaryColor": "#e96232",
      "searchBackgroundColor": "#999999",
      "secondaryColor": "#cacaca",
      "splashBackgroundColor": "#21202e",
      "successColor": "#008214",
      "unFocusedIconColor": "#cccccc",
    }
  }
>
  TEST
</Text>
`;

exports[`StyledBaseScroll renders correctly 1`] = `
<RCTScrollView
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "height": "100%",
        "width": "100%",
      },
    ]
  }
  theme={
    Object {
      "backgroundColor": "#fff",
      "backgroundColorLight": "rgb(250, 250, 250)",
      "badgeBackgroundColors": Object {
        "launch": "#dcf0f0",
        "wiki": "#e4e4e4",
        "youtube": "#c93a2f",
      },
      "badgeFontColors": Object {
        "launch": "#090909",
        "wiki": "#4c4c4c",
        "youtube": "#fefcff",
      },
      "failColor": "#990000",
      "focusedIconColor": "#e96232",
      "fontColors": Object {
        "light": "#fff",
        "primary": "#e96232",
        "regular": "#090909",
        "subtle": "#666",
      },
      "headerBackgroundColor": "#fff",
      "listItemBackgroundColor": "rgb(252,252,252)",
      "mainFont": "#090909",
      "placeHolderFontColor": "#cccccc",
      "primaryColor": "#e96232",
      "searchBackgroundColor": "rgb(252,252,252)",
      "secondaryColor": "#666",
      "splashBackgroundColor": "#21202e",
      "successColor": "#008214",
      "unFocusedIconColor": "#cccccc",
    }
  }
>
  <View />
</RCTScrollView>
`;

exports[`SubText renders correctly 1`] = `
<Text
  fontColor="subtle"
  fontType="regular"
  size="s"
  style={
    Array [
      Object {
        "color": "#666",
        "fontFamily": "RobotoCondensed-Regular",
        "fontSize": 12,
        "lineHeight": 13,
      },
    ]
  }
  theme={
    Object {
      "backgroundColor": "#fff",
      "backgroundColorLight": "rgb(250, 250, 250)",
      "badgeBackgroundColors": Object {
        "launch": "#dcf0f0",
        "wiki": "#e4e4e4",
        "youtube": "#c93a2f",
      },
      "badgeFontColors": Object {
        "launch": "#090909",
        "wiki": "#4c4c4c",
        "youtube": "#fefcff",
      },
      "failColor": "#990000",
      "focusedIconColor": "#e96232",
      "fontColors": Object {
        "light": "#fff",
        "primary": "#e96232",
        "regular": "#090909",
        "subtle": "#666",
      },
      "headerBackgroundColor": "#fff",
      "listItemBackgroundColor": "rgb(252,252,252)",
      "mainFont": "#090909",
      "placeHolderFontColor": "#cccccc",
      "primaryColor": "#e96232",
      "searchBackgroundColor": "rgb(252,252,252)",
      "secondaryColor": "#666",
      "splashBackgroundColor": "#21202e",
      "successColor": "#008214",
      "unFocusedIconColor": "#cccccc",
    }
  }
>
  TEST
</Text>
`;

exports[`Title renders correctly 1`] = `
<Text
  fontColor="regular"
  fontType="bold"
  size="l"
  style={
    Array [
      Object {
        "color": "#090909",
        "fontFamily": "RobotoCondensed-Bold",
        "fontSize": 16,
        "lineHeight": 24,
      },
    ]
  }
  theme={
    Object {
      "backgroundColor": "#fff",
      "backgroundColorLight": "rgb(250, 250, 250)",
      "badgeBackgroundColors": Object {
        "launch": "#dcf0f0",
        "wiki": "#e4e4e4",
        "youtube": "#c93a2f",
      },
      "badgeFontColors": Object {
        "launch": "#090909",
        "wiki": "#4c4c4c",
        "youtube": "#fefcff",
      },
      "failColor": "#990000",
      "focusedIconColor": "#e96232",
      "fontColors": Object {
        "light": "#fff",
        "primary": "#e96232",
        "regular": "#090909",
        "subtle": "#666",
      },
      "headerBackgroundColor": "#fff",
      "listItemBackgroundColor": "rgb(252,252,252)",
      "mainFont": "#090909",
      "placeHolderFontColor": "#cccccc",
      "primaryColor": "#e96232",
      "searchBackgroundColor": "rgb(252,252,252)",
      "secondaryColor": "#666",
      "splashBackgroundColor": "#21202e",
      "successColor": "#008214",
      "unFocusedIconColor": "#cccccc",
    }
  }
>
  TEST
</Text>
`;
